src/serialize/numpy.rs

```diff
@@ -972,6 +1022,22 @@ impl Serialize for NumpyUint64 {
     }
 }
 
+#[repr(C)]
+pub struct NumpyFloat16 {
+    ob_refcnt: Py_ssize_t,
+    ob_type: *mut PyTypeObject,
+    value: f16,
+}
+
+impl Serialize for NumpyFloat16 {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {   
+        self.value.to_f32().serialize(serializer)
```

Comment:
there should be a better way to do it, but I can't figure out something better

Reply:
It does, unfortunately. It's more complex than I expected. You can close this PR.